ovs/extensions/healthcheck/volumedriver/volumedriver_health_check.py

```diff
@@ -55,29 +55,43 @@ def check_dtl(logger):
         :type logger: ovs.log.healthcheck_logHandler.HCLogHandler
         """
         test_name = "check_dtl"
-
+        test_output = {
+            "failure": [],
+            "warning": [],
+            "success": []
+        }
+        vdisk_name = None
+        vdisk_dtl = None
         # Fetch vdisks hosted on this machine
         if len(VolumedriverHealthCheck.MACHINE_DETAILS.vdisks_guids) != 0:
+            # Force refresh the cache
+            VolumedriverHealthCheck.MACHINE_DETAILS.invalidate_dynamics("vdisks_guids")
             for vdisk_guid in VolumedriverHealthCheck.MACHINE_DETAILS.vdisks_guids:
                 try:
-                    results = VolumedriverHealthCheck._check_disk_dtl(vdisk_guid=vdisk_guid)
-                    if results[1] == "ok_standalone":
-                        logger.warning("VDisk {0}'s DTL is disabled".format(results[0]), test_name)
-                    elif results[1] == "ok_sync":
-                        logger.success("VDisk {0}'s DTL is enabled and running.".format(results[0]), test_name)
-                    elif results[1] == "degraded":
-                        logger.failure("VDisk {0}'s DTL is degraded.".format(results[0]), test_name)
-                    elif results[1] == "catchup" or results[1] == "catch_up":
-                        logger.warning("VDisk {0}'s DTL is enabled but still syncing.".format(results[0]), test_name)
+                    vdisk_name, vdisk_dtl = VolumedriverHealthCheck._check_disk_dtl(vdisk_guid=vdisk_guid)
+                    if vdisk_dtl == "ok_standalone":
+                        test_output["warning"].append((vdisk_name, "VDisk {0}'s DTL is disabled".format(vdisk_name)))
+                        logger.warning("VDisk {0}'s DTL is disabled".format(vdisk_name))
+                    elif vdisk_dtl == "ok_sync":
+                        test_output["success"].append((vdisk_name, "VDisk {0}'s DTL is enabled and running.".format(vdisk_name)))
+                        logger.success("VDisk {0}'s DTL is enabled and running.".format(vdisk_name))
+                    elif vdisk_dtl == "degraded":
+                        test_output["failure"].append((vdisk_name, "VDisk {0}'s DTL is degraded.".format(vdisk_name)))
+                        logger.failure("VDisk {0}'s DTL is degraded.".format(vdisk_name))
+                    elif vdisk_dtl == "catchup" or vdisk_dtl == "catch_up":
+                        test_output["warning"].append((vdisk_name, "VDisk {0}'s DTL is enabled but still syncing.".format(vdisk_name)))
+                        logger.warning("VDisk {0}'s DTL is enabled but still syncing.".format(vdisk_name))
                     else:
-                        logger.warning("VDisk {0}'s DTL has an unknown status: {1}.".format(results[0], results[1]),
-                                       test_name)
+                        test_output["warning"].append((vdisk_name, "VDisk {0}'s DTL has an unknown status: {1}.".format(vdisk_name, vdisk_dtl)))
```

Comment:
An unknown status should be ```failure``` instead of ```warning```, because most likely the StorageDriver is down at this point